src/test/java/emissary/core/HDMobileAgentBugTest.java

```diff
@@ -0,0 +1,205 @@
+        assertTrue(att2.transformHistory().contains(FAKE_SKIPPED_HISTORY));
+        assertTrue(att2.transformHistory().contains(FAKE_VERIFY_OTHER_HISTORY));
+
+        /*
+         * Unexpected Results - att1 and att2 are processed together. Att1 goes to the expected two places:
+         * FakeAnalyzePlace, and FakeVerifyPlace. Att2 is expected to go to three places: FakeAnalyzePlace,
+         * FakeSkippedPlace, and FakeVerifyPlace but FakeSkippedPlace is never called
+         *
+         * Explanation: The bug is in HDMobileAgent on line 244, the comment reads
+         *   // Add any other payload that has the same current form
+         *   // and last place visited as this one while we are here
+         *   for (final IBaseDataObject slug : this.payloadList) { ...
+         *
+         * Att1 and Att2 have the same current form, FRM-PROCESSED-alternative, so they get grouped together. There are
+         * no analyze places left for FRM-PROCESSED-alternative, so they go to the Verify stage - FakeVerifyPlace.
+         * Att2 has another form, FRM-PROCESSED-OTHER, that is never processed.
+         *
+         * filename: testing_file-att-1
+         * creationTimestamp: Fri Nov 09 18:49:28 GMT 2018
+         * currentForms: [FRM-PROCESSED-alternative, FRM-PROCESSED]
+         * filetype: TST
+         * transform history (3) :
+         *   *.FAKE_ANALYZE.ANALYZE.http://localhost:8543/VisitedHereFirstPlace.localhost:8543
+         *   FRM-PROCESSED-alternative.FAKE_ANALYZE.ANALYZE.http://localhost:8543/FakeAnalyzePlace$5050
+         *   FRM-PROCESSED-alternative.FAKE_VERIFY.VERIFY.http://localhost:8543/FakeVerifyPlace$5050
+         *
+         * filename: testing_file-att-2
+         * creationTimestamp: Fri Nov 09 18:49:28 GMT 2018
+         * currentForms: [FRM-PROCESSED-alternative, FRM-PROCESSED-OTHER]
+         * filetype: TST
+         * transform history (3) :
+         *   *.FAKE_ANALYZE.ANALYZE.http://localhost:8543/VisitedHereFirstPlace.localhost:8543
+         *   FRM-PROCESSED-alternative.FAKE_ANALYZE.ANALYZE.http://localhost:8543/FakeAnalyzePlace$5050
+         *   FRM-PROCESSED-alternative.FAKE_VERIFY.VERIFY.http://localhost:8543/FakeVerifyPlace$5050
+         */
+        att1 = generateIbdo(1, form1);
+        att2 = generateIbdo(2, form2);
+        ma.clear();
+        ma.addPayload(att1);
+        ma.addPayload(att2);
+        ma.agentControl(analyze);
+        assertTrue(att1.transformHistory().contains(FAKE_ANALYZE_HISTORY));
+        assertTrue(att1.transformHistory().contains(FAKE_VERIFY_HISTORY));
+        assertTrue(att2.transformHistory().contains(FAKE_ANALYZE_HISTORY));
+        // wrong
+        assertTrue(att2.transformHistory().contains(FAKE_VERIFY_HISTORY));
+        // these should be part of the history, but are not
+        assertTrue(!att2.transformHistory().contains(FAKE_SKIPPED_HISTORY));
+        assertTrue(!att2.transformHistory().contains(FAKE_VERIFY_OTHER_HISTORY));
```

Comment:
these assertions could be simplified to `assertFalse()`, but is not per se necessary

Reply:
We can certainly clean this up in a follow-on. There are a few changes I'd like to see made in this test.